src/search/Form.js

```diff
@@ -0,0 +1,34 @@
+import { React, Component } from 'react';
+import './Form.css';
+
+class Form extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchInput: ''
+    }
+  }
+
+  handleChange = event => {
+    this.setState({ [event.target.name]: event.target.value}, () => {
+      this.props.filterMovies(this.state.searchInput)
+    });
+  }
```

Comment:
Nice work finding such an elegant solution to this problem. So the SetState method can accept a callback function? Which in this case means it won't change the component's state until filterMovies is finished executing? Might need you to walk me through it one more time.